File: _CodeTopics/LeetCode_contest/weekly/weekly2020/215/215_1.py
class OrderedStream(object):

    def __init__(self, n):
        """
        :type n: int
        """
        
        self.ptr = 1
        self.arr = [0] * (n+1)

    def insert(self, id, value):
        """
        :type id: int
        :type value: str
        :rtype: List[str]
        """
        
        self.arr[id] = value
        if self.arr[self.ptr] == 0:
            return []
        start = self.ptr
        while self.ptr <= len(self.arr)-1 and self.arr[self.ptr] != 0:
            self.ptr += 1
        return self.arr[start:self.ptr]


# Your OrderedStream object will be instantiated and called as such:
# obj = OrderedStream(n)
# param_1 = obj.insert(id,value)

"""
https://leetcode-cn.com/submissions/detail/123516424/

101 / 101 个通过测试用例
状态：通过
执行用时: 616 ms
内存消耗: 13.3 MB
"""

File: _CodeTopics/LeetCode/1601-1800/001656--weekly215_1/tran/215_1.cpp
// 第 215 场周赛第一题
class OrderedStream {
public:
    int ptr;
    vector<string> arr;
public:
    OrderedStream(int n) {
        ptr = 1;
        arr.assign(n+1, "");
    }
    
    vector<string> insert(int idKey, string value) {
        arr[idKey] = value;
        if (arr[ptr] == "") {
            return vector<string>{};
        }
        int start = ptr;
        while (ptr <= arr.size()-1 && arr[ptr] != "") {
            ++ptr;
        }
        return vector<string>(arr.begin()+start, arr.begin()+ptr);
    }
};


/**
 * Your OrderedStream object will be instantiated and called as such:
 * OrderedStream* obj = new OrderedStream(n);
 * vector<string> param_1 = obj->insert(idKey,value);
 */

/*
https://leetcode.cn/submissions/detail/350886066/

执行用时：
112 ms
, 在所有 C++ 提交中击败了
42.63%
的用户
内存消耗：
79 MB
, 在所有 C++ 提交中击败了
100.00%
的用户
通过测试用例：
101 / 101
*/
